src/plugins/conditionals/conditionals.c

```diff
@@ -168,140 +220,155 @@ static int evalCondition (const char * leftSide, Comparator cmpOp, const char *
 	}
 	snprintf (lookupName, len, "%s/%s", keyName (parentKey), leftSide);
 	key = ksLookupByName (ks, lookupName, 0);
-	if (!key)
+	if (cmpOp == NEX)
 	{
-		ELEKTRA_SET_ERRORF (133, parentKey, "Key %s not found but is required for the evaluation of %s", lookupName, condition);
-		result = -2;
+		if (key)
+			result = 0;
+		else
+			result = 1;
+		goto Cleanup;
+	}
+	if (!key && cmpOp != OR && cmpOp != AND)
+	{
+		if (!keyGetMeta (parentKey, "error"))
+		{
+			ELEKTRA_SET_ERRORF (133, parentKey, "Key %s not found but is required for the evaluation of %s", lookupName,
+					    condition);
+		}
+		result = 0;
 		goto Cleanup;
 	}
 	long ret;
-	ret = compareStrings (keyString (key), compareTo);
+	if (cmpOp == OR || cmpOp == AND)
+		ret = compareStrings (leftSide, rightSide, NULL);
+	else
+		ret = compareStrings (keyString (key), compareTo, suffixList);
 	switch (cmpOp)
 	{
 	case EQU:
-		if (!ret)
-		{
-			result = 1;
-		}
+		if (!ret) result = 1;
 		break;
 	case NOT:
-		if (ret)
-		{
-			result = 1;
-		}
+		if (ret) result = 1;
 		break;
 	case LT:
-		if (ret < 0)
-		{
-			result = 1;
-		}
+		if (ret < 0) result = 1;
 		break;
 	case LE:
-		if (ret <= 0)
-		{
-			result = 1;
-		}
+		if (ret <= 0) result = 1;
 		break;
 	case GT:
-		if (ret > 0)
-		{
-			result = 1;
-		}
+		if (ret > 0) result = 1;
 		break;
 	case GE:
-		if (ret >= 0)
-		{
-			result = 1;
-		}
+		if (ret >= 0) result = 1;
 		break;
 	case SET:
 		keySetString (key, compareTo);
 		result = 1;
 		break;
+	case AND:
+		if (ret == 0 && !strcmp (leftSide, "'1'")) result = 1;
+		break;
+	case OR:
+		if (!strcmp (leftSide, "'1'") || !strcmp (rightSide, "'1'")) result = 1;
+		break;
 	default:
 		result = -1;
 		break;
 	}
 // freeing allocated heap
 Cleanup:
-	if (lookupName)
-	{
-		elektraFree (lookupName);
-	}
-	if (compareTo)
-	{
-		elektraFree (compareTo);
-	}
+	if (lookupName) elektraFree (lookupName);
+	if (compareTo) elektraFree (compareTo);
 	return result;
 }
 
 
-static int parseSingleCondition (const char * condition, KeySet * ks, Key * parentKey)
+static int parseSingleCondition (const char * condition, const Key * suffixList, KeySet * ks, Key * parentKey)
 {
 	Comparator cmpOp;
 	char * opStr;
 	if ((opStr = strstr (condition, "==")))
-	{
```

Comment:
please keep the `{}`. I think you can easily refactor this part to a condition2cmpOp function.